cmd/operator-sdk/generate/csv.go

```diff
@@ -191,25 +182,28 @@ func validateVersion(version string) error {
 	return nil
 }
 
-func writeCRDsToDir(crdPaths []string, toDir string) error {
-	for _, p := range crdPaths {
-		b, err := ioutil.ReadFile(p)
-		if err != nil {
-			return err
-		}
-		typeMeta, err := k8sutil.GetTypeMetaFromBytes(b)
-		if err != nil {
-			return err
-		}
-		if typeMeta.Kind != "CustomResourceDefinition" {
-			continue
-		}
+// findCRDFileSet searches all directories and files in path for CRD manifests,
+// returning a map of paths to file contents.
+func findCRDFileSet(path string) (map[string][]byte, error) {
+	crdFileSet := map[string][]byte{}
+	info, err := os.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("crd's must be read from a directory. %s is a file", path)
+	}
 
-		path := filepath.Join(toDir, filepath.Base(p))
-		err = ioutil.WriteFile(path, b, fileutil.DefaultFileMode)
+	crdPaths, err := k8sutil.GetCRDManifestPaths(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, crdPath := range crdPaths {
+		b, err := ioutil.ReadFile(crdPath)
```

Comment:
If we're not including a `crd` input key in this PR, we should ignore any CRD's coming from `<deploy dir>/olm-catalog`, since they can have the same name as one in `<deploy dir>/crds` with different content.

Reply:
Hmm, I think I might just add a `crd` input key while we're here if it's not a big change to remove the ambiguity.
And to guard against re-reading CRDs in any case, I'm going to explicitly ignore `olm-catalog/...` in the `filepath.Walk()` since that's always a constant no matter the inputs/outputDir.

Reply:
Added a `--crd-dir` flag and the filepath Walks for CRD manifests now ignore the olm directory.